### DISCUSSION TITLE
Migration Help from v6 to v7

### DISCUSSION THREAD

**Discussion by dakshsaraf** [react-hook-form/react-hook-form#4894] [Q&A]
I am trying to migrate from v6 to v7. I am using a lot of MUI components for the inputs and am not sure how to modify the functionality of the register and Controller component to work with. Can you please provide me some suggestions for this. I read the change guide on the website but am a little confused.

Here is a code sandbox link that was working with the following versions.
@hookform/devtools: 2.2.1
@hookform/resolvers: 0.1.1
react-hook-form: 6.15.5

and I am trying to update to 
@hookform/devtools: 3.0.2
@hookform/resolvers: 2.4.0
react-hook-form: 7.2.1

https://codesandbox.io/s/react-hook-form-v7-migration-9f2o7?file=/src/App.js

**Comment by bluebill1049** [comment]
If you find migrate to v7 is overwhelming, feel safe to stay in v6 as it is still getting maintained.

could of tip i can summarise:

```ts
// v6
<Textinput name="test" inputRef={register} /> 


// v7
const {...rest, ref} = register('test');
<Textinput {...rest} inputRef={ref} />  

// v6
<Controller as={<TextInput />} />
<Controller render={(props) => <TextInput {...props} />} />

// v7
<Controller render={({ field, fieldState, formState }) => <TextInput {...field} />} /> // v6
```
